app/src/main/java/com/digitalhouse/dhwallet/HomeCardFragment.kt

```diff
@@ -56,8 +56,7 @@ class HomeCardFragment : Fragment(R.layout.fragment_home_card) {
 
         val viewPager = view.findViewById<ViewPager2>(R.id.list_card)
         viewPager.adapter = CardViewPageAdapter(this, listCards){
-            value = it.limit
-            numCard = it.number
+            card = it
             sendToDetail()
```

Comment:
Aqui é mais como sugestão, para vc não precisar ter uma variável mutável como é o caso do `card`, você poderia passar o `it` direto como parâmetro do seu `sendToDetail()`, ficando `sendToDetail(it)`...

ou se vc quiser arriscar usar um lambda expression, seu adapter ficaria assim:
`viewPager.adapter = CardViewPageAdapter(this, listCards, ::sendToDetail)`